server/channels/store/sqlstore/job_store.go

```diff
@@ -172,34 +172,84 @@ func (jss SqlJobStore) UpdateStatus(id string, status string) (*model.Job, error
 	return job, nil
 }
 
-func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus string, newStatus string) (bool, error) {
+func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus string, newStatus string) (*model.Job, error) {
+	if jss.DriverName() == model.DatabaseDriverMysql {
+		tx, err := jss.GetMaster().Beginx()
+		if err != nil {
+			return nil, errors.Wrap(err, "begin_transaction")
+		}
+		defer finalizeTransactionX(tx, &err)
+
+		builder := jss.getQueryBuilder().
+			Update("Jobs").
+			Set("LastActivityAt", model.GetMillis()).
+			Set("Status", newStatus).
+			Where(sq.Eq{"Id": id, "Status": currentStatus})
+
+		if newStatus == model.JobStatusInProgress {
+			builder = builder.Set("StartAt", model.GetMillis())
+		}
```

Comment:
I guess this is how it's been implemented so far, but are we okay with potentially overriding a job's start time? It feels like we are making some business logical assumptions.

Reply:
This would be the start time (when the job moved from `pending` to `in_progress`).

Reply:
Yeah, I just don't think this decision should happen at the store layer, but that's a different story. Nothing to do here.

Reply:
@streamer45 right, I agree -- we should pass it in.